common/functions/src/scalars/maths/pow.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright 2020 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+
+use common_datavalues::prelude::*;
+use common_datavalues::DataSchema;
+use common_datavalues::DataType;
+use common_exception::ErrorCode;
+use common_exception::Result;
+use num::traits::Pow;
+
+use crate::scalars::function_factory::FunctionDescription;
+use crate::scalars::function_factory::FunctionFeatures;
+use crate::scalars::Function;
+
+#[derive(Clone)]
+pub struct PowFunction {
+    display_name: String,
+}
+
+impl PowFunction {
+    pub fn try_create(display_name: &str) -> Result<Box<dyn Function>> {
+        Ok(Box::new(PowFunction {
+            display_name: display_name.to_string(),
+        }))
+    }
+
+    pub fn desc() -> FunctionDescription {
+        FunctionDescription::creator(Box::new(Self::try_create))
+            .features(FunctionFeatures::default().deterministic())
+    }
+}
+
+impl Function for PowFunction {
+    fn name(&self) -> &str {
+        &*self.display_name
+    }
+
+    fn num_arguments(&self) -> usize {
+        2
+    }
+
+    fn return_type(&self, args: &[DataType]) -> Result<DataType> {
+        if is_numeric(&args[0]) || args[0] == DataType::String || args[0] == DataType::Null {
+            Ok(DataType::Float64)
+        } else {
+            Err(ErrorCode::IllegalDataType(format!(
+                "Expected numeric, but got {}",
+                args[0]
+            )))
+        }
+    }
+
+    fn nullable(&self, _input_schema: &DataSchema) -> Result<bool> {
+        Ok(false)
+    }
+
+    fn eval(&self, columns: &DataColumnsWithField, input_rows: usize) -> Result<DataColumn> {
+        let x_column: &DataColumn = &columns[0].column().cast_with_type(&DataType::Float64)?;
+        let y_column: &DataColumn = &columns[1].column().cast_with_type(&DataType::Float64)?;
+
+        let result = match (x_column, y_column) {
+            (DataColumn::Array(x_series), DataColumn::Constant(y, _)) => {
+                if y.is_null() {
+                    DFFloat64Array::full_null(input_rows)
+                } else {
+                    let y: f64 = DFTryFrom::try_from(y.clone())?;
+                    let opt_iter = x_series.f64()?.into_iter().map(|x| x.map(|x| x.pow(y)));
```

Comment:
Sorry, I forgot to tell you that we can use `ArrayApply`here to improve the performance.

Reply:
The reason is that we take validity at last to fit the cpu cache.

Reply:
> The reason is that we take validity at last to fit the cpu cache.

TIL and updated.